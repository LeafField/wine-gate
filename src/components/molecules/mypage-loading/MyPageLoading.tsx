import React, { FC } from "react";
import WineTitleLoading from "../../atoms/wine-title-loading/WineTitleLoading";

const MyPageLoading: FC = () => {
  return (
    <main className="max-w-[64.9375rem] space-y-4 px-4 main tablet:w-[calc(var(--vw)-401px)] tablet:px-0">
      {Array.from({ length: 10 }).map((_, i) => (
        <WineTitleLoading key={i} />
      ))}
    </main>
  );
};

export default MyPageLoading;
