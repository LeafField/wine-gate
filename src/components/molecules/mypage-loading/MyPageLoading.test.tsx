import { screen, render } from "@testing-library/react";
import MyPageLoading from "./MyPageLoading";

describe("MyPageLoadingの単体テスト", () => {
  test("WineTitleLoadingコンポーネントが10個レンダリングされているか", () => {
    expect.assertions(10);
    render(<MyPageLoading />);
    const loaders = screen.getAllByText("読み込み中...");
    loaders.forEach((dom) => {
      expect(dom).toBeInTheDocument();
    });
  });

  test("スナップショットテスト", () => {
    const { container } = render(<MyPageLoading />);
    expect(container).toMatchSnapshot();
  });
});
