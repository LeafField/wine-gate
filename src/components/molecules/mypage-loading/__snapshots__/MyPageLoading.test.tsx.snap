// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPageLoadingの単体テスト スナップショットテスト 1`] = `
<div>
  <main
    class="max-w-[64.9375rem] space-y-4 px-4 main tablet:w-[calc(var(--vw)-401px)] tablet:px-0"
  >
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
    <div
      class="h-[4.375rem] max-w-[64.9375rem] animate-pulse bg-gray tablet:w-[calc(var(--vw)-401px)]"
    >
      <span
        class="sr-only"
      >
        読み込み中...
      </span>
    </div>
  </main>
</div>
`;
